src/Umbraco.Deploy.Contrib/Migrators/ReplaceMatryoshkaArtifactMigrator.cs

```diff
@@ -0,0 +1,100 @@
+using System.Linq;
+using System;
+using Umbraco.Cms.Core.Deploy;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Semver;
+using Umbraco.Deploy.Core.Migrators;
+using Umbraco.Deploy.Infrastructure.Artifacts;
+using Umbraco.Deploy.Infrastructure.Artifacts.ContentType;
+using System.Collections.Generic;
+using Umbraco.Extensions;
+
+namespace Umbraco.Deploy.Contrib.Migrators;
+
+/// <summary>
+/// Migrates the <see cref="DataTypeArtifact" /> to remove Matryoshka data types (using the <see cref="EditorAlias" /> editor) and
+/// <see cref="ContentTypeArtifactBase" /> to replace properties using the removed Matryoshka data types with native groups.
+/// </summary>
+public sealed class ReplaceMatryoshkaArtifactMigrator : ArtifactMigratorBase<IArtifact>
+{
+    private const string EditorAlias = "Our.Umbraco.Matryoshka.GroupSeparator";
+
+    private readonly HashSet<Guid> _removedDataTypeKeys = [];
```

Comment:
Artifact migrators are registered as singletons, so this might contain removed data type keys from multiple imports and keep them in memory for the rest of the application runtime (until the application is restarted).

However, since these are GUIDs, they take up an insignificant amount of memory and the chances they conflict with other data types is realistically zero (and a restart would otherwise fix it).

Tracking they keys is required, because new data types are only saved in process pass 0 (which happens after all artifacts are migrated before the initial pass), so simply assuming non-existent data types are the removed Matryoshka ones doesn't work...